x-pack/plugins/maps/public/reducers/map.js

```diff
@@ -437,7 +442,7 @@ function updateWithDataResponse(state, action) {
 
   dataRequest.data = action.data;
   dataRequest.dataMeta = { ...dataRequest.dataMetaAtStart, ...action.meta };
-  dataRequest.dataMetaAtStart = null;
+  delete dataRequest.dataMetaAtStart;
```

Comment:
is this needed? If not, then just keep as is but modify https://github.com/elastic/kibana/blob/dc013cb80f4256f7ffbcb8b814efee8590fc4a8a/x-pack/plugins/maps/common/descriptor_types/data_request_descriptor_types.d.ts#L68  to allow null . Assigning null is much cleaner than deleting properties on an object, which is pretty advanced JS-usage anyway.